### DISCUSSION TITLE
Fringe region

### DISCUSSION THREAD

**Discussion by udhaya-chandiran** [Exawind/amr-wind#1188] [Q&A] [answered]
Is there a way to set up a fringe region (a Rayleigh damping layer with a time-varying input velocity) in the streamwise/spanwise direction in AMR-Wind?

**Comment by udhaya-chandiran** [comment]
Any thoughts on this would be helpful.

**Comment by mbkuhn** [comment, marked as answer]
Hello! This isn't a current capability in AMR-Wind. The primary method of time-varying input velocity is to use the boundary-plane approach, having generated the boundary-plane data in a precursor simulation. However, if you're willing to get in the code, the necessary pieces for this feature are there. 

In amr-wind/equation_systems/icns/source_terms/RayleighDamping.cpp, the Rayleigh Damping source term is set up to operate at the upper boundary in z. Without many changes, that could be tweaked to operate at the lower boundary in x. 

Also, the Rayleigh Damping currently only takes a constant reference velocity. However, there are a few other source terms that can read in a time table of reference values. GeostrophicForcing.cpp, BodyForce.cpp, and ABLForcing.cpp would be good references for you. 

I'm not sure how this feature will interact with the inflow boundary condition, and there may be some other wrinkles like that. But this is a fairly direct path to getting the approach you are looking for.

**Comment by udhaya-chandiran** [comment]
Thank you for the response. My intended simulations include wind farm-induced gravity waves, which will interact with the boundary conditions at the inlet, so using fringe regions at the inlet is the only way known to me to get correct results.

Thank you for detailing the modifications needed. I don't know C++ and have limited experience modifying source codes, so I will need time and support to make these tweaks. I will see if I can run my simulations using a different solver. If that is not possible, I will consider making these modifications.
